server/chat-commands/admin.ts

```diff
@@ -34,6 +34,12 @@ function bash(command: string, context: CommandContext, cwd?: string): Promise<[
 	});
 }
 
+const BLACKLISTED_PROPS = ['valueOf', 'constructor'];
+
+function isWhitelistedProp(prop: string) {
+	return !(prop.includes('__') || prop.toLowerCase().includes('prop') || BLACKLISTED_PROPS.includes(prop));
+}
```

Comment:
You should be using `hasOwnProperty`, not a literal blacklist.

Reply:
That doesn't seem to solve what i want, here?

Reply:
Hm, I think I see. The whitelist should be documented. I'll suggest a change once you move `listMethods` over.

Reply:
I also don't see a need for you to check for functions specifically. Do you know why you did that?

Reply:
I don't know if prototypes include non-functions? Nor do i know how it interacts, so I went for safety.

Reply:
Prototypes can include anything. In practice they shouldn't include non-functions (and the `class` syntax disallows non-functions), but they're supported by JS and should be copied if they do exist.

Reply:
Ah, alright.